pkg/cloudprovider/provider/kubevirt/provider.go

```diff
@@ -845,37 +846,21 @@ func getAffinity(config *Config, matchKey, matchValue string) *corev1.Affinity {
 	return affinity
 }
 
-func hostnameWeightedAffinityTerm(matchKey, matchValue string) []corev1.WeightedPodAffinityTerm {
-	return []corev1.WeightedPodAffinityTerm{
-		{
-			Weight: 1,
-			PodAffinityTerm: corev1.PodAffinityTerm{
-				LabelSelector: &metav1.LabelSelector{
-					MatchLabels: map[string]string{
-						matchKey: matchValue,
-					},
-				},
-				TopologyKey: topologyKeyHostname,
-			},
-		},
+func addPrefixToSecondaryDisk(secondaryDisks []SecondaryDisks, prefix string) {
+	for i := range secondaryDisks {
+		secondaryDisks[i].Name = fmt.Sprintf("%s-%s", prefix, secondaryDisks[i].Name)
 	}
 }
 
-func hostnameAffinityTerm(matchKey, matchValue string) []corev1.PodAffinityTerm {
-	return []corev1.PodAffinityTerm{
-		{
-			LabelSelector: &metav1.LabelSelector{
-				MatchLabels: map[string]string{
-					matchKey: matchValue,
-				},
-			},
-			TopologyKey: topologyKeyHostname,
-		},
+func getTopologySpreadConstraints(config *Config, matchLabels map[string]string) []corev1.TopologySpreadConstraint {
+	if len(config.TopologySpreadConstraints) != 0 {
+		for i := range config.TopologySpreadConstraints {
+			config.TopologySpreadConstraints[i].LabelSelector = &metav1.LabelSelector{MatchLabels: matchLabels}
+		}
+		return config.TopologySpreadConstraints
 	}
-}
-
-func addPrefixToSecondaryDisk(secondaryDisks []SecondaryDisks, prefix string) {
-	for i := range secondaryDisks {
-		secondaryDisks[i].Name = fmt.Sprintf("%s-%s", prefix, secondaryDisks[i].Name)
+	// Return default TopologySpreadConstraint
+	return []corev1.TopologySpreadConstraint{
+		{MaxSkew: 1, TopologyKey: topologyKeyHostname, WhenUnsatisfiable: corev1.ScheduleAnyway, LabelSelector: &metav1.LabelSelector{MatchLabels: matchLabels}},
```

Comment:
just a nit, could you fix the formatting of this structure? It will be more readable if it's not a one liner.